### PR TITLE
zig cc: add linker -Map flag

### LINKED ISSUE #18356: Support linker map files in `zig cc`
### Zig Version

0.11.0

### Steps to Reproduce and Observed Behavior

Here's a fragment of my actual command (obtained from `VERBOSE=1 make`)

```
$ clang -O2 -g -DNDEBUG -flto=thin -Wl,--export-dynamic -rdynamic  -Xlinker -Map=skrouterd.map/usr/bin/ld: cannot open map file skrouterd.map: Permission denied
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Clang tries to access the map file (and if the file actually existed, it would succeed)

When I run the same compilation with `zig cc` instead of clang, I get an error from commandline argument parser

```
$ zig cc -O2 -g -DNDEBUG -flto=thin -Wl,--export-dynamic -rdynamic  -Xlinker -Map=skrouterd.map
error: unsupported linker arg: -Map=skrouterd.map
```

### Expected Behavior

Please support the `-Map=` linker argument. I can try looking into this myself if given some initial pointers on what needs to be done. Thanks.

### PR DESCRIPTION
**Pull request by jiridanek**
Linker maps are a diagnostic/debugging feature. It causes the linker to write out memory ranges (similar to the runtime content of `/proc/self/map` on Linux) and some such info. I need this because a program I want to compile with `zig cc` uses this.

Fixes #18356

### CODE REVIEW

File: src/main.zig
Comment (jiridanek) [inline src/main.zig:2402]: This handles `-Map=file` which is enough for me. Should `-Map file` be handled as well?

File: src/Compilation.zig
Comment (jiridanek) [inline src/Compilation.zig:1187]: I'm not sure what I'm actually doing here and whether I should do this.

**Review by kubkon — CHANGES_REQUESTED** [review]
What is the purpose of this flag in an ELF linker? Additionally, before we consider this for merging, we need a test in `test/link/elf.zig` with `llvm=true` and `lld=true`. Once we move away from lld as the default linker, this will help in narrowing down the missing functionality.

**Comment by jiridanek** [comment]
@kubkon I could not figure out how to check the `.map` file content from inside of the test. More importantly, i haven't figured out how to run the test I just wrote (or this test _and_ the least amount of other tests) on my machine

```
% stage4/bin/zig test ../test/tests.zig --test-filter  testMapFlag
All 0 tests passed.
stage4/bin/zig test ../test/tests.zig --test-filter testMapFlag  4.76s user 1.58s system 100% cpu 6.333 total
```

Can you or anyone else please suggest how to do both correctly?

**Comment by andrewrk** [comment]
`zig build test-link`

You can pass `--help` to see more options (the help menu is augmented by the contents of build.zig scripts so it's different in different projects).

**Comment by andrewrk** [comment]
Please see https://github.com/ziglang/zig/issues/18356#issuecomment-1877618990. @kubkon will be the final word on this, but I think it would be better to not even decide whether to add this functionality until some other things are done. Until then, if you need this feature you will need to use a different toolchain than zig cc. @kubkon please close this pull request if you agree.

**Comment by kubkon** [comment]
I have to agree with @andrewrk here - this is a new feature request that while may look innocent and easy to implement with LLD in mind, it needs some careful consideration if ported to Zig's linker. Until we decide if we want to accept your proposal, I am closing this PR as not applicable.

**Pull request closed without merging**
